safe/sample.py

```diff
@@ -636,6 +637,90 @@ def scaffold_decoration(
                 )
         return total_sequences
 
+    def pattern_decoration(
+        self,
+        scaffold: Union[str, dm.Mol],
+        n_samples_per_trial: int = 10,
+        n_trials: int = 1,
+        do_not_fragment_further: bool = True,
+        sanitize: bool = False,
+        random_seed: Optional[int] = None,
+        add_dot: bool = True,
+        n_scaff_random: Optional[int] = 3,
+        n_scaff_samples: Optional[int] = 10,
+        scaff_temperature: float = 1.0,
+        **kwargs: Optional[Dict[Any, Any]],
+    ) -> List[str]:
+        """
+        Perform pattern decoration using the pretrained SAFE model. The pattern decoration algorithm works by first examplifying the patterns
+        as a set of scaffold then performing scaffold decoration on each scaffold.
+
+        !!! warning
+            Designing molecules from a given molecule pattern is more challenging than fragment-constrained design.
+            SAFE does not currently support complex SMARTS pattern schemes (e.g., valence or connectivity constraints, some ring constraints).
+            This function works best when sampling given a list of atoms. However, sampling depends on the model's conditional probabilities,
+            meaning that if the model assigns zero probability to a token, you are unlikely to see it.
+
+        Args:
+            scaffold: Scaffold (with attachment points) to decorate.
+            n_samples_per_trial: Number of new molecules to generate for each randomization.
+            n_trials: Number of randomizations to perform.
+            do_not_fragment_further: Whether to prevent further fragmentation of the scaffold.
+            sanitize: Whether to sanitize the generated molecules and ensure the scaffold is present.
+            random_seed: Seed for randomization.
+            n_scaff_random: Number of scaffold randomizations to try (to reposition constraints in the string and increase rollout likelihood).
+                Increasing this will improve sampling, but will require more time.
+            n_scaff_samples: Maximum number of samples to sample for a given scaffold from the pattern.
+                Increasing this will make sure you have more diversity in the scaffold coming from the pattern
+            scaff_temperature: Temperature to use when sampling valid scaffolds from the pattern.
```

Comment:
Maybe add more intuitive explanation here, something like " As the temperature increases, the generated scaffolds and structures become less confident but more diverse."